experimental/LibSingWrapper.jl

```diff
@@ -0,0 +1,244 @@
+import Singular.LibSing: *
+
+export milnor, tjurina, global_milnor_number, global_tjurina_number
+
+### The Milnor algebra of an isolated hypersurface singularity f
+
+@Markdown.doc """
+    milnor(f::MPolyElem; point::Vector=[])
+
+For an isolated hypersurface singularity `f` at a point `point`, 
+compute the triple `(gb, μ, g)` where `gb` is a standard basis for 
+the jacobian ideal, `μ` is the Milnor number, and `g` is a ``𝕜``-base 
+for the Milnor algebra.
+
+The default argument for `point` is the origin.
+"""
+function milnor(f::MPolyElem; point::Vector{T}=Vector{elem_type(base_ring(parent(f)))}()) where {T}
+  iszero(f) && error("polynomial is zero")
+  R = parent(f)
+  kk = coefficient_ring(f)
+  all((x->(parent(x) == kk)), point) || return milnor(f, point=kk.(point))
+
+  # We can not provide a full default vector of zeroes in the 
+  # signature of the function, because we would need a full 
+  # instance of the field at that point. Hence, we have to 
+  # do that at a later point here.
+  k_point = kk.(point)
+  if length(point) != 0
+    # check for the right number of coordinates
+    length(k_point) == ngens(R) || error("vector does not have the correct length")
+  else
+    # fill up the default otherwise
+    k_point = elem_type(kk)[zero(kk) for i in 1:ngens(R)]
+  end
+
+  # Set up the localization at `point` so that we have 
+  # the local orderings at our disposal
+  W = Localization(MPolyComplementOfKPointIdeal(R, k_point))
+
+  # compute the jacobian ideal
+  Df = jacobi_matrix(f)::AbstractAlgebra.Generic.MatSpaceElem{typeof(f)}
+  I = ideal(R, [Df[i, 1] for i in 1:nrows(Df)])
+  
+  # transfer to the localized ring since otherwise we could not work with 
+  # local orderings
+  WI = W(I)
+  # compute a standard basis w.r.t a local ordering 
+  stdI = groebner_basis(WI)
+
+  # carry out the computations on the Singular side
+  Singular.dimension(singular_gens(stdI)) == 0 || error("singularity is not isolated")
+  milnor_number = Singular.vdim(singular_gens(stdI)) 
+  # make the conversion to the Oscar side 
+  milnor_kbase = [to_oscar_side(stdI, g) for g in gens(Singular.kbase(singular_gens(stdI)))]
+  return (stdI, milnor_number, milnor_kbase)
```

Comment:
In julia as opposed to Singular, you'd in general return three values not a triple. The use is the same, but notionally here and else where you have 3 return values, not a single triple. You'd want to use them independently I assume...